e2e/nomostest/gitproviders/repository.go

```diff
@@ -115,6 +116,8 @@ type Repository struct {
 	Logger *testlogger.TestLogger
 
 	yamlSerializer *jserializer.Serializer
+
+	defaultWaitTimeout time.Duration
```

Comment:
nit: since it can't be overridden here, it's not really "Default". It's more like "retryTimeout" or "pushRetryTimeout"

Reply:
otoh, "defaultWaitTimeout" makes it sorta obvious it's linked to that e2e setting.